File: server/data/results/translations/anchored/go/balanced_parens.go
package main

import "fmt"

func is_balanced(text string) bool {
	var stack []rune
	pairs := map[rune]rune{
		')': '(',
		']': '[',
		'}': '{',
	}
	for _, ch := range text {
		if ch == '(' || ch == '[' || ch == '{' {
			stack = append(stack, ch)
		} else if opening, ok := pairs[ch]; ok {
			if len(stack) == 0 {
				return false
			}
			top := stack[len(stack)-1]
			stack = stack[:len(stack)-1]
			if top != opening {
				return false
			}
		}
	}
	return len(stack) == 0
}

func main() {
	samples := []string{"()", "([{}])", "(]", "((()", "{[()]}"}
	for _, sample := range samples {
		if is_balanced(sample) {
			fmt.Println("true")
		} else {
			fmt.Println("false")
		}
	}
}

File: server/data/results/translations/anchored/cpp/balanced_parens.cpp
#include <iostream>
#include <string>
#include <vector>
#include <unordered_map>

bool is_balanced(const std::string& text) {
    std::vector<char> stack;
    std::unordered_map<char, char> pairs = {
        {')', '('},
        {']', '['},
        {'}', '{'}
    };
    for (char ch : text) {
        if (ch == '(' || ch == '[' || ch == '{') {
            stack.push_back(ch);
        } else if (pairs.find(ch) != pairs.end()) {
            if (stack.empty()) {
                return false;
            }
            char top = stack.back();
            stack.pop_back();
            if (top != pairs[ch]) {
                return false;
            }
        }
    }
    return stack.empty();
}

int main() {
    std::vector<std::string> samples = {"()", "([{}])", "(]", "((()", "{[()]}"};
    for (const auto& sample : samples) {
        if (is_balanced(sample)) {
            std::cout << "true\n";
        } else {
            std::cout << "false\n";
        }
    }
    return 0;
}